extensions/bitwarden/src/hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { getLocalStorageItem, showToast, ToastStyle, removeLocalStorageItem, setLocalStorageItem } from "@raycast/api";
+import { useState, useEffect } from "react";
+import { BitwardenApi } from "./api";
+
+const LocalStorageSessionKey = "sessionToken"
+export function useBitwarden(bitwardenApi: BitwardenApi): [string | null | undefined, (sessionToken: string | null) => void] {
+  const [sessionToken, setSessionToken] = useState<string | null>();
+
+  useEffect(() => {
+    async function getSessionToken() {
+      const sessionToken = await getLocalStorageItem<string>(LocalStorageSessionKey);
+
+      const status = await bitwardenApi.getVaultStatus(sessionToken)
+
+      if (status === "unlocked") setSessionToken(sessionToken);
+      else if (status === "locked") setSessionToken(null);
+      else if (status === "unauthenticated") {
+        try {
+          const toast = await showToast(ToastStyle.Animated, "Login in...", "It may takes some time");
+          await bitwardenApi.login()
+          toast.hide();
+          setSessionToken(null);
+        } catch (error) {
+          showToast(ToastStyle.Failure, "An error occurred during login!", "Please check your crendentials");
+        }
+      }
+    }
+    getSessionToken();
+  }, []);
+
+  return [
+    sessionToken,
+    async (sessionToken: string | null) => {
+      if (!sessionToken) {
+        removeLocalStorageItem(LocalStorageSessionKey);
+        setSessionToken(null);
+      } else {
+        setLocalStorageItem(LocalStorageSessionKey, sessionToken);
+        setSessionToken(sessionToken);
+      }
```

Comment:
Two things here:
1. In general, it's a best practice to avoid "if not" patterns. I'd rather this look like:
    ```ts
    if (sessionToken) {
      setLocalStorageItem(LocalStorageSessionKey, sessionToken);
      setSessionToken(sessionToken);
    } else {
      removeLocalStorageItem(LocalStorageSessionKey);
      setSessionToken(null);
    }
    ```
1. In the case where we have a `sessionToken`, is it necessary to both `setLocalStorageItem` and `setSessionToken`? I think the ergonomics would be improved if we didn't have to worry about setting the local storage item when setting a new `sessionToken`, because `setSessionToken` just handled that for us. Still, I'd think no changes were actually necessary, since we're confirming that things are already in a good state? In the case where we don't have a `sessionToken`, it would be nice to call a `clearSessionToken` helper that takes care of unsetting the local variable(s) and the local storage entry. With those edits, this becomes dramatically simpler:
    ```ts
    if (!sessionToken) {
      clearSessionToken()
    }
    ```

Reply:
We need to call both function to both:
- change the local storage value (setLocalStorageItem)
- rerender the component tree (setSessionToken)

I don't understand you other comments, I feel like this helper is quite easy to use donwstream:

```javascript
const [sessionToken, setSessionToken] = useSessionToken(bitwardenApi);
setSessionToken("session-token") // This will both change the localStorage value and rerender the component tree
```